snuba/migrations/snuba_migrations/events/0011_rebuild_errors.py

```diff
@@ -0,0 +1,185 @@
+]
+
+sample_expr = "cityHash64(toString(event_id))"
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    This migration rebuilds the errors table, with the following changes:
+
+    - Dropped org_id column
+    - Dropped event_string column
+    - Dropped event_hash materialized column
+    - Dropped primary_hash_hex materialized column
+    - Primary key is now more in line with what we have for the events table:
+        - no org_id
+        - cityHash64(toString(event_id)) instead of primary_hash_hex, event_hash
+    - Sharding key for distributed table is also cityHash64(toString(event_id))
+    - No _tags_flattened and _contexts_flattened since we are mostly no longer
+    using these already
+    - TTL reflects retention_days value not only 30 or 90
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.EVENTS,
+                table_name="errors_local_new",
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    storage_set=StorageSetKey.EVENTS,
+                    version_column="deleted",
+                    order_by="(project_id, toStartOfDay(timestamp), %s)" % sample_expr,
+                    partition_by="(retention_days, toMonday(timestamp))",
+                    sample_by=sample_expr,
+                    ttl="timestamp + toIntervalDay(retention_days)",
+                    settings={"index_granularity": "8192"},
+                ),
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.EVENTS,
+                table_name="errors_local_new",
+                column=Column(
+                    "_tags_hash_map",
+                    Materialized(Array(UInt(64)), TAGS_HASH_MAP_COLUMN),
+                ),
+                after="tags",
+            ),
+            operations.DropTable(
+                storage_set=StorageSetKey.EVENTS, table_name="errors_local",
+            ),
+            operations.RenameTable(
+                storage_set=StorageSetKey.EVENTS,
+                old_table_name="errors_local_new",
+                new_table_name="errors_local",
+            ),
+        ]
+
+    def backwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.DropTable(
+                storage_set=StorageSetKey.EVENTS, table_name="errors_local_new"
+            )
+        ]
+
+    def forwards_dist(self) -> Sequence[operations.Operation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.EVENTS,
+                table_name="errors_dist_new",
+                columns=columns,
+                engine=table_engines.Distributed(
+                    local_table_name="errors_local", sharding_key=sample_expr,
+                ),
+            ),
+            operations.DropTable(
```

Comment:
There won't be a _tags_hash_map column on the distributed table this way. Is that intentional ?